src/utils/actions.js

```diff
@@ -8,228 +8,339 @@ export const getValueList = async (app, values, isDate) => {
     translation: 'Object.ActionButton.UnlockAllSelections',
     group: 'selection',
-    getActionCall: ({ app }) => async () => {
-      await app.unlockAll();
-    },
+    getActionCall:
+      ({ app }) =>
+        async () => {
+          await app.unlockAll();
+        },
     requiredInput: [],
   },
   {
     value: 'unlockField',
     translation: 'Object.ActionButton.UnlockAField',
     group: 'selection',
-    getActionCall: ({ app, qStateName, field }) => async () => {
-      if (field) {
-        const fieldObj = await app.getField(field, qStateName);
-        await fieldObj.unlock();
-      }
-    },
+    getActionCall:
+      ({ app, qStateName, field }) =>
+        async () => {
+          if (field) {
+            const fieldObj = await app.getField(field, qStateName);
+            await fieldObj.unlock();
+          }
+        },
     requiredInput: ['field'],
   },
   {
     value: 'setVariable',
     translation: 'Object.ActionButton.SetVariable',
     group: 'variables',
-    getActionCall: ({ app, variable, value }) => async () => {
-      if (variable && value) {
-        try {
-          const variableObj = await app.getVariableByName(variable);
-          await variableObj.setStringValue(value);
-        } catch (e) {
-          // no-op
-        }
-      }
-    },
+    getActionCall:
+      ({ app, variable, value }) =>
+        async () => {
+          if (variable && value) {
+            try {
+              const variableObj = await app.getVariableByName(variable);
+              await variableObj.setStringValue(value);
+            } catch (e) {
+            // no-op
+            }
+          }
+        },
     requiredInput: ['variable', 'value'],
   },
   {
     value: 'doReload',
     translation: 'Object.ActionButton.DoReload',
     group: 'reload',
-    getActionCall: ({ app, partial }) => async () => {
-      const e = await app.doReload(0, !!partial, false);
-      if (e) {
-        await app.doSave();
-      }
-    },
+    getActionCall:
+      ({ app, partial }) =>
+        async () => {
+          const e = await app.doReload(0, !!partial, false);
+          if (e) {
+            await app.doSave();
+          }
+        },
     requiredInput: ['partial'],
   },
+  {
+    /** *************************************
+     * Execute Automation contacts internal urls to obtain automation
+     * information and execute the automation selected in the property panel in
+     * edit mode.
+     *
+     * ARGS
+     * app - Reference to current app inherited from index.js
+     * automation - the item id of the automation to contact the items api
+     * automationPostData - boolean value. If true, creates a bookmark of the
+     * current selections and sends the resulting bookmark id as an input
+     * parameter to the selected automation.
+     */
+
+    translation: 'Object.ActionButton.ExecuteAutomation',
+    value: 'executeAutomation',
+    getActionCall:
+      ({ app, automation, automationPostData }) =>
```

Comment:
`app.id` and `automation` comes from properties/layout and cannot be trusted, so they need to be properly escaped when used in URLS and payloads.

Reply:
Okay. So we should probably add some sanitiser to the button as well. We never concluded that discussion on how to do it right? Copy of a solution or should we start looking into a library that we can use?

Reply:
Hi there, let me know how I can help here. I'm ready to jump back in.